ci_framework/roles/os_net_setup/tasks/main.yml

```diff
@@ -0,0 +1,37 @@
+# assiming having a login session with the right namespace already
+- name: get openstack admin credentials from k8s
+  block:
+  - name: Get clouds.yaml configuration
+    ansible.builtin.command:
+      cmd: oc get configmaps openstack-config -o jsonpath={.data}
+    register: clouds_yaml
+    environment:
+      PATH: "{{ path }}"
+
+  - name: Get the OSP admin password
+    ansible.builtin.command:
+      cmd: oc get secrets openstack-config-secret -o jsonpath={.data}
+    register: osp_secret
+    environment:
+      PATH: "{{ path }}"
+  vars:
+    default_path: "{{ ansible_user_dir }}/.crc/bin:{{ ansible_user_dir }}/.crc/bin/oc:{{ ansible_user_dir }}/bin:{{ ansible_env.PATH }}"
+    path: "{{ cifmw_path | default(default_path) }}"
+
+- name: Construct auth param for openstack
+  ansible.builtin.set_fact:
+    openstack_auth: |
+      {{ ((( clouds_yaml.stdout_lines| join('\n') | from_json)['clouds.yaml']|from_yaml)['clouds']['default'] |
+      combine ({'auth': { 'password' : ((osp_secret.stdout_lines|join('\n')| from_json) ['secure.yaml']|b64decode|from_yaml)['clouds']['default']['auth']['password']} }, recursive=true))['auth'] }}
+    region_name: "{{ ((clouds_yaml.stdout_lines|join('\n')|from_json)['clouds.yaml']|from_yaml)['clouds']['default']['region_name'] }}"
+
+- openstack.cloud.networks_info:
+    auth: "{{ openstack_auth }}"
+  register: net_info
+
+- name: Process network list element
+  ansible.builtin.include_tasks:
```

Comment:
please consider using `ansible.builtin.import_tasks` instead - unless there's a loop somewhere.

More info: https://ci-framework.readthedocs.io/en/latest/development/01_guidelines.html#include-vs-import

Reply:
There is loop two lines below - so include_tasks should stay then?

Reply:
the contrary - import_* are for "static" imports, meaning pre-compute (so faster consumption) - include_* are for dynamic load and multiple-includes